src/components/constraints/ConstraintsPanel.svelte

```diff
@@ -227,10 +284,32 @@
     <GridMenu {gridSection} title="Constraints" />
     <PanelHeaderActions status={$constraintsStatus} indeterminate>
       <PanelHeaderActionButton
-        disabled={$simulationStatus !== Status.Complete}
+        title="Re-Check"
+        disabled={$simulationStatus !== Status.Complete || $constraintsStatus !== Status.Complete}
+        tooltipContent="Re-check constraints"
+        showLabel
+        use={[
+          [
+            permissionHandler,
+            {
+              hasPermission: $plan
+                ? featurePermissions.constraintRuns.canCreate(user, $plan, $plan.model) && !$planReadOnly
+                : false,
+              permissionError: $planReadOnly
+                ? PlanStatusMessages.READ_ONLY
+                : 'You do not have permission to run constraint checks',
+            },
+          ],
+        ]}
+        on:click={() => $plan && effects.checkConstraints($plan, user, true)}
+      >
+        <RefreshIcon />
+      </PanelHeaderActionButton>
+      <PanelHeaderActionButton
+        disabled={$simulationStatus !== Status.Complete || $constraintsStatus === Status.Complete}
```

Comment:
Should a user be allowed to check constraints if an enabled constraint is missing a required parameter?
![image](https://github.com/user-attachments/assets/d44b5144-8e77-4632-b81b-df7ff8439a65)

Reply:
I think in terms of good UX, no they shouldn't. But introducing that kind of check would be out of scope for this initial effort. If we already did something similar with running simulations, then maybe it would have been feasible this cycle.

Reply:
Not sure I've seen simulations with required args, might exist, but either way fine by me to punt on this.